StringCalculatorTest/StringCalculationTest.cs

```diff
@@ -1,13 +1,180 @@
+using kata_string_calculator.Delimiters;
+using kata_string_calculator.Validation;
 using Xunit;
 
 namespace StringCalculatorTest
 {
-
-    public class StringCalculatorTest
+    public class StringCalculationTests
     {
+        private readonly StringCalculator _stringCalculator;
+        public StringCalculationTests()
+        {
+            var delimitersHandler = new DelimitersHandler();
+            var validator = new NumberValidation();
+            _stringCalculator = new StringCalculator(delimitersHandler, validator);
+        }
+        
+        [Fact]
+        public void Should_Return_A_Number_When_Given_A_String()
+        {
+            // Act
+            var result = _stringCalculator.Add("");
+
+            // Assert
+            Assert.IsType<int>(result);
+        }
+        
         [Fact]
-        public void Test1()
+        public void Should_Return_Zero_When_Given_An_Empty_String()
+        {
+            // Act
+            var result = _stringCalculator.Add("");
+
+            // Assert
+            Assert.Equal(decimal.Zero, result);
+        }
+        
+        [Theory]
+        [InlineData(1)]
+        [InlineData(2)]
+        [InlineData(3)]
+        public void Should_Return_The_Same_Number_That_Was_Passed_In(int expectedNumber)
+        {
+            // Act
+            var actualNumber = _stringCalculator.Add($"{expectedNumber}");
+
+            // Assert
+            Assert.Equal(expectedNumber, actualNumber);
+        }
+        
+        [Theory]
+        [InlineData(3, "1,2")]
+        [InlineData(66, "24,42")]
+        [InlineData(44, "39,5")]
+        public void Should_Return_The_Sum_When_Two_Numbers_Are_Passed_In(int expectedSum, string stringInput)
+        {
+            // Act
+            var actualSum = _stringCalculator.Add(stringInput);
+
+            // Assert
+            Assert.Equal(expectedSum, actualSum);
+        }
+        
+        [Theory]
+        [InlineData(6, "1,2,3")]
+        [InlineData(20, "3,5,3,9")]
+        [InlineData(27, "4,6,2,8,7")]
+        public void Should_Return_The_Sum_When_Multiple_Numbers_Are_Passed_In(int expectedSum, string stringInput)
+        {
+            // Act
+            var actualSum = _stringCalculator.Add(stringInput);
+
+            // Assert
+            Assert.Equal(expectedSum, actualSum);
+        }
+        
+        [Theory]
+        [InlineData(6, "1,2\n3")]
+        [InlineData(20, "3\n5\n3,9")]
+        [InlineData(27, "4\n6,2,8\n7")]
+        public void Should_Return_The_Sum_When_Multiple_Numbers_Are_Passed_In_With_Both_Comma_And_NewLineCharacter_As_Separators(int expectedSum, string stringInput)
+        {
+            // Act
+            var actualSum = _stringCalculator.Add(stringInput);
+
+            // Assert
+            Assert.Equal(expectedSum, actualSum);
+        }
+
+        [Theory]
+        [InlineData(3, "//;\n1;2")]
+        [InlineData(8, "//-\n3-5")]
+        [InlineData(7, "//+\n6+1")]
```

Comment:
```suggestion
        [InlineData(7, "//+\n6+1")]
        [InlineData(10, "//+\n6+1/n3")]
```
Maybe not necessary, but I'll probably add one more case that has a different pattern in order to prove "All existing steps should still be supported".